### PR TITLE
Feat/remove isolated vertices

### LINKED ISSUE #43: Remove isolated vertices
Add a function to remove vertices that are used in any elements

### PR DESCRIPTION
**Pull request by panquez**
fix #43

### CODE REVIEW

File: src/geode/mesh/builder/polyhedral_solid_builder.cpp [hunk 1]
```diff
@@ -486,6 +486,22 @@ namespace geode
         do_delete_polyhedra( to_delete );
     }
 
+    template < index_t dimension >
+    void PolyhedralSolidBuilder< dimension >::delete_isolated_vertices()
+    {
+        std::vector< bool > to_delete( polyhedral_solid_.nb_vertices(), false );
+        for( auto v : geode::Range{ polyhedral_solid_.nb_vertices() } )
```
Comment (BotellaA) [inline src/geode/mesh/builder/polyhedral_solid_builder.cpp]: geode::

File: src/geode/mesh/builder/polyhedral_solid_builder.cpp [hunk 2]
```diff
@@ -486,6 +486,22 @@ namespace geode
...
+        {
+            const auto& polyhedron_vertex =
+                polyhedral_solid_.polyhedron_around_vertex( v );
+            if( polyhedron_vertex.polyhedron_id == geode::NO_ID )
```
Comment (BotellaA) [inline src/geode/mesh/builder/polyhedral_solid_builder.cpp]: geode::

File: src/geode/mesh/builder/polygonal_surface_builder.cpp [hunk 3]
```diff
@@ -414,6 +414,23 @@ namespace geode
...
+    template < index_t dimension >
+    void PolygonalSurfaceBuilder< dimension >::delete_isolated_vertices()
+    {
+        std::vector< bool > to_delete(
+            polygonal_surface_.nb_vertices(), false );
+        for( auto v : geode::Range{ polygonal_surface_.nb_vertices() } )
+        {
+            const auto& polygon_vertex =
+                polygonal_surface_.polygon_around_vertex( v );
+            if( polygon_vertex.polygon_id == geode::NO_ID )
```
Comment (BotellaA) [inline src/geode/mesh/builder/polygonal_surface_builder.cpp]: geode::

File: src/geode/mesh/builder/polygonal_surface_builder.cpp [hunk 4]
```diff
@@ -414,6 +414,23 @@ namespace geode
         do_delete_polygons( to_delete );
     }
 
+    template < index_t dimension >
+    void PolygonalSurfaceBuilder< dimension >::delete_isolated_vertices()
+    {
+        std::vector< bool > to_delete(
+            polygonal_surface_.nb_vertices(), false );
+        for( auto v : geode::Range{ polygonal_surface_.nb_vertices() } )
```
Comment (BotellaA) [inline src/geode/mesh/builder/polygonal_surface_builder.cpp]: geode::

Comment (panquez) [inline src/geode/mesh/builder/polygonal_surface_builder.cpp]:
```suggestion
        for( auto v : Range{ polygonal_surface_.nb_vertices() } )
```

File: src/geode/mesh/builder/polygonal_surface_builder.cpp (hunk 3 above)
Comment (panquez) [inline src/geode/mesh/builder/polygonal_surface_builder.cpp]:
```suggestion
            if( polygon_vertex.polygon_id == NO_ID )
```

**Pull request merged**